api/src/test/java/org/openmrs/module/locationbasedaccess/aop/PersonSearchAdviserTest.java

```diff
@@ -0,0 +1,213 @@
+import org.openmrs.module.locationbasedaccess.aop.interceptor.PersonServiceInterceptorAdvice;
+import java.util.List;
+import java.util.Set;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertEquals;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+public class PersonSearchAdviserTest extends AOPContextSensitiveTest {
+
+	private UserService userService;
+	private PersonService personService;
+	private LocationService locationService;
+	private static final String XML_FILENAME_WITH_PERSON_DATA = "include/PatientTestData.xml";
+	private static final String XML_FILENAME_WITH_ADMIN_USER_DATA = "include/AdminUserData.xml";
+
+	/** Configured values in the demo dataset **/
+	private static final int DEMO_PERSON_ATTRIBUTE_TYPE_ID = 10;
+	private static final int DEMO_LOCATION1_ID = 1;
+	private static final int DEMO_LOCATION2_ID = 2;
+	private static final int DEMO_USER1_ID = 6001;
+
+	@Override
+	protected void setInterceptorAndServices(TestWithAOP testCase) {
+		testCase.setInterceptor(PersonServiceInterceptorAdvice.class);
+		testCase.addService(PersonService.class);
+	}
+
+	@Before
+	public void setUp()  throws Exception {
+		userService = Context.getUserService();
+		personService = Context.getPersonService();
+		locationService = Context.getLocationService();
+		executeDataSet(XML_FILENAME_WITH_ADMIN_USER_DATA);
+		executeDataSet(XML_FILENAME_WITH_PERSON_DATA);
+	}
+
+	@Test
+	public void getPeople_adminShouldGetPeopleFromAllLocations() {
+		Context.authenticate("admin", "test");
+		User authenticatedUser = Context.getAuthenticatedUser();
+		assertNotNull(authenticatedUser);
+		assertTrue(authenticatedUser.isSuperUser());
+
+		Location location1 = locationService.getLocation(DEMO_LOCATION1_ID);
+		Location location2 = locationService.getLocation(DEMO_LOCATION2_ID);
+
+		List<Person> personList = personService.getPeople("Patient", false);
+		assertEquals(personList.size(), 3);
+		assertNull(personList.get(0).getAttribute(DEMO_PERSON_ATTRIBUTE_TYPE_ID));
+		assertEquals(personList.get(1).getAttribute(DEMO_PERSON_ATTRIBUTE_TYPE_ID).getValue(), location1.getUuid());
+		assertEquals(personList.get(2).getAttribute(DEMO_PERSON_ATTRIBUTE_TYPE_ID).getValue(), location2.getUuid());
+	}
+
+	@Test
+	public void getPeople_nonAdminUserWithOutLocationPropertyShouldNotGetPersonInformation() {
+		User normalUser = userService.getUser(DEMO_USER1_ID);
+		Context.authenticate(normalUser.getUsername(), "userServiceTest");
+		assertFalse(normalUser.isSuperUser());
+
+		assertEquals(normalUser.getUserProperties().size(), 0);
+
+		try {
+			List<Person> personList = personService.getPeople("Patient", false);
+		} catch (APIAuthenticationException e) {
+			assertThat(e.getMessage(), is("error.privilegesRequired"));
```

Comment:
Actually, I'm not really sure the code works correctly at all. Look, I've tried adding tests when a user has location properties but is not an admin. Every time I've tried to get person from the location he has access to, it returned me error.privilegesRequired. Same problem both with `getPeople()` and with `getPersonByUuid()` and with `getPerson()`. I just didn't know where to mention this thing.
This is why I cannot be sure this is caused by missing location property for the user and not by something else. Can you comment on this maybe?

Reply:
So then, you have to add the privileges to that user through the XML configuration file which you have attached at the beginning of the class.  Explore some other projects (Better in Core) to get the XML configuration to add privileges.
Here, we won't check about the privileges. So you can give any privilege to that user, and then check for the person for the location.

Reply:
Check here to add privileges to the testing user : https://github.com/openmrs/openmrs-module-locationbasedaccess/blob/master/api/src/test/resources/include/PatientTestData.xml#L37-L48

Reply:
@suthagar23 I'm facing a problem while adding a privilege to PatientTestData.xml. I add the following two lines:
```
<privilege privilege="Get People" description="This is a test privilege" uuid="d979d066-15e6-467c-9d4b-cb575ef97f6u"/>
<role_privilege role="userRole" privilege="Get People"/>
```
But after `mvn clean package` they are removed. Could you give me any hint on why does it happen, and how to address the issue?

Reply:
What do you mean by removed after mvn clean? It can't happen. You can see, there are some privileges already added to that user role. Like that, you need to add this privilege also.

Reply:
Can you try : `mvn clean install` ?